From 92aef139765a3b9a1e1fa826d2b688071da34c19 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 21 Sep 2022 12:46:31 +0300
Subject: [PATCH 04/30] s32: clock: Fix compilation warning

This fixes the warning:
error: comparison of integer expressions of different signedness: 'int'
and 'size_t' {aka 'unsigned int'}

Issue: ALB-9298
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/nxp/s32/clk/clk.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/nxp/s32/clk/clk.c b/drivers/nxp/s32/clk/clk.c
index 2cd3efcbf..129b6963c 100644
--- a/drivers/nxp/s32/clk/clk.c
+++ b/drivers/nxp/s32/clk/clk.c
@@ -122,7 +122,7 @@ int get_clk(uint32_t drv_id, uint32_t clk_id, struct clk *clock)
 }
 
 
-static int process_parents_prop(int *parent_index, size_t nparents,
+static int process_parents_prop(size_t *parent_index, size_t nparents,
 				const fdt32_t *parents, struct clk *clk,
 				struct clk *parent_clk)
 {
@@ -166,10 +166,10 @@ int dt_clk_apply_defaults(void *fdt, int node)
 {
 	const fdt32_t *clocks, *parents, *rates;
 	int nclocks_size, nparents_size, nrates_size;
-	int ret, index, parent_index;
+	int ret, index;
 	uint32_t clk_drv_id, clk_id, freq, i;
 	struct clk clk, parent_clk;
-	size_t nparents;
+	size_t nparents, parent_index;
 	int fret = 0;
 
 	clocks = fdt_getprop(fdt, node, "assigned-clocks", &nclocks_size);
-- 
2.17.1

